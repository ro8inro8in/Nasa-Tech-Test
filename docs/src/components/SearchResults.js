import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchResults.css";

const SearchResults = ({results}) => {
  return results ? (
    results.map((result) => {
      return (
        <>
          <div className="image-container">
            <img
              className="image"
              src={result}
              alt="Search result from NASA database"
              data-testid="image"
            />
          </div>
        </>
      );
    })
  ) : (
    <p>Sorry, no results. Please try again</p>
    );
};

SearchResults.propTypes = {
  results: PropTypes.array.isRequired
};

export default SearchResults;
