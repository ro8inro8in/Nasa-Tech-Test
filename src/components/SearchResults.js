import React from "react";
import PropTypes from "prop-types";
import "../styles/App.css";

const SearchResults = ({ results }) => {
  !results.length ? (
    <p>No Results</p>
  ) : (
    results.map((image) => (
      <img className="card-image" src={image} alt="spaceImage" />
    ))
  );
};

export default SearchResults;

SearchResults.PropTypes = {
results: PropTypes.string.isRequired
}


