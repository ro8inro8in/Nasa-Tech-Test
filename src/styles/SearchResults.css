.search-results {
    column-count: 4;
    column-gap: 15px;
    margin: 50px;
}
.image-container {
    display: inline-block;
    width: 100%;
    margin-top: 15px;
}
.image-container img {
    display: block;
    width: 100%;
    border-radius: 25px;
}
.search-results > p {
  column-span: all;
}

@media only screen and (max-width: 750px) {
  .search-results {
      column-gap: 5px;
      margin: 25px;
  }
  
  .image-container {
      margin-top: 5px;
  }

  .image-container img {
      border-radius: 5px;
  }
}