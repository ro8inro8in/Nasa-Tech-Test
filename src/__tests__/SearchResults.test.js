import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";


describe("Search Results", () => {
    const { asFragment } = render(<SearchResults />);
  
    it("renders correctly", () => {
      expect(asFragment()).toMatchSnapshot();
    });
  });