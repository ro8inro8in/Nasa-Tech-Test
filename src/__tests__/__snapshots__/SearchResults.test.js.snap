// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results renders correctly 1`] = `
<DocumentFragment>
  <p>
    Search Results
  </p>
  <img
    alt="planet-image"
    class="planet-image"
    src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
  />
</DocumentFragment>
`;
